widgetssdk/src/main/java/com/glia/widgets/helper/ViewExtensions.kt

```diff
@@ -145,3 +150,86 @@ internal val View.layoutInflater: LayoutInflater get() = LayoutInflater.from(thi
 internal fun TextView.setCompoundDrawableTintListCompat(tint: ColorStateList?) {
     TextViewCompat.setCompoundDrawableTintList(this, tint)
 }
+
+internal fun TextView.setLocaleText(@StringRes stringKey: Int, vararg values: StringKeyPair) {
+    registerLocaleListener(stringKey, *values) { upToDateTranslation ->
+        text = upToDateTranslation
+    }
+}
+
+internal fun TextView.setText(locale: LocaleString?) {
+    if (locale == null) {
+        text = ""
+        return
+    }
+    registerLocaleListener(locale.stringKey, *locale.values.toTypedArray()) { upToDateTranslation ->
+        text = upToDateTranslation
+    }
+}
+
+internal fun TextView.setLocaleHint(@StringRes stringKey: Int, vararg values: StringKeyPair) {
+    registerLocaleListener(stringKey, *values) { upToDateTranslation ->
+        hint = upToDateTranslation
+    }
+}
+
+internal fun TextView.setHint(locale: LocaleString?) {
+    if (locale == null) {
+        hint = ""
+        return
+    }
+    registerLocaleListener(locale.stringKey, *locale.values.toTypedArray()) { upToDateTranslation ->
+        hint = upToDateTranslation
+    }
+}
+
+internal fun View.setLocaleContentDescription(@StringRes stringKey: Int, vararg values: StringKeyPair) {
+    registerLocaleListener(stringKey, *values) { upToDateTranslation ->
+        contentDescription = upToDateTranslation
+    }
+}
+
+internal fun View.setContentDescription(locale: LocaleString?) {
+    if (locale == null) {
+        contentDescription = ""
+        return
+    }
+    registerLocaleListener(locale.stringKey, *locale.values.toTypedArray()) { upToDateTranslation ->
+        contentDescription = upToDateTranslation
+    }
+}
+
+internal fun Toolbar.setLocaleNavigationContentDescription(@StringRes stringKey: Int, vararg values: StringKeyPair) {
+    registerLocaleListener(stringKey, *values) { upToDateTranslation ->
+        navigationContentDescription = upToDateTranslation
+    }
+}
+
+private fun View.registerLocaleListener(@StringRes stringKey: Int, vararg values: StringKeyPair, listener: (String) -> Unit) {
+    val localeManager = Dependencies.getLocaleProvider()
+    val localeListener = {
+        // It is strange that it should already be UI thread according to:
+        // https://github.com/salemove/android-sdk/blob/4dbf361ecf108b478b997e415734240778af2e41/androidSdk/src/main/java/com/glia/androidsdk/internal/locale/LocaleManagerInternal.kt#L71
+        // But without `runOnUiThread` it sometimes does not work
+        context.asActivity()?.runOnUiThread {
+            listener.invoke(localeManager.getString(stringKey, values.toList()))
+        }
+    }
+    localeListener.invoke()
+    val disposable = localeManager.getLocaleObservable().subscribe(
+        { localeListener.invoke() },
```

Comment:
What do you think about this solution, it is simpler and will avoid listener invocation in case of the same value.
```suggestion
   val disposable = localeManager.getLocaleObservable()
        .startWithItem("stub")
        .map { localeManager.getString(stringKey, values.toList()) }
        .distinctUntilChanged()
        .subscribe { post { listener(it) } }
```

Reply:
Nice one 👍